sdk/servicebus/Azure.Messaging.ServiceBus/src/Processor/SessionReceiverManager.cs

```diff
@@ -127,8 +127,11 @@ private async Task CreateAndInitializeSessionReceiver(CancellationToken processo
 
         private async Task CreateReceiver(CancellationToken processorCancellationToken)
         {
+            bool releaseSemaphore = false;
             try
             {
+                await _concurrentAcceptSessionsSemaphore.WaitAsync(processorCancellationToken).ConfigureAwait(false);
```

Comment:
Does it make sense to attempt to acquire synchronously here first with a `Wait(0)` and then fall back to async?

Reply:
Yeah good catch - I can also add this back to the other semaphore wait in ServiceBusProcessor.